src/v3/src/transformer/field/transform.ts

```diff
@@ -81,7 +81,8 @@ export const transformStepInputs = (
       const flattenedInputs = flattenInputs(input);
       return [...acc, ...flattenedInputs];
     }, [])
-    .filter((input) => input.visible !== false && input.mutable !== false)
+    .filter((input) => input.visible === true
+        || (input.visible !== false && input.mutable !== false))
```

Comment:
When fields are visible but immutable, we still need to transform them for display. This is what this update handles.